ros2doctor/ros2doctor/command/doctor.py

```diff
@@ -65,7 +65,8 @@ def main(self, *, parser, args):
         if args.report_failed and fail != 0:
             fail_reports = generate_reports(categories=fail_category)
             for report_obj in fail_reports:
-                format_print(report_obj)
+                if report_obj is not None:
```

Comment:
I don't see what this check is needed for. Can you please provide a reproducible example when this is needed.

Reply:
The easiest way to reproduce this is to build a workspace, source it, `unset ROS_DISTRO`, and then run `ros2 doctor -r`.  This somewhat artificially causes the `package.py` check to fail, and it to place a `None` value in the list of Report objects.

But actually, you are right.  Looking into it further, this actually highlighted a different bug in platform.py.  See 99d2d9b for a fix there.